### ISSUE TITLE
Is this repo still alive?  Should my fork take over?

### ISSUE THREAD

**Issue by chrisant996** [vladimir-kotikov/clink-completions#152]
@vladimir-kotikov do you have interest in continuing to maintain this repo?

Would it be helpful for me to take over (e.g. grant me write access), or would it be helpful for me to position [my fork](https://github.com/chrisant996/clink-completions) as the new active repo?  PR #149 has a lot of enhancements, and some are significant.

**Comment by vladimir-kotikov** [comment]
Hey, Chris! This is something I had been proposing to Cmder folks [quite some time ago](https://github.com/cmderdev/cmder/issues/1843). Nothing really changed since that time - I have neither enough time nor, more important, personal interest in contributing to this project anymore, but I don't want it to be abandoned completely so I'd appreciate if you take over the maintenance. Feel free to continue developing and maintaining this repo (I'll add you to the members of the repo) or your own fork - whichever you think is more appropriate for you, just let me know about what you decide.

And thank you for your initiative! I really appreciate that someone will continue building tools that help other people!

**Comment by chrisant996** [comment]
@vladimir-kotikov That works very well, thanks Vladimir. I'm happy to accept the responsibility, and I'm grateful for all the effort you've invested in this project over the years!

I like the idea of keeping this the main repo.  It means less work for others (especially Cmder).

**Comment by DRSDavidSoft** [comment]
I appreciate your efforts towards keeping clink completions great as always!  Looking forward to seeing adb.exe completions in the official repo in the future

**Comment by chrisant996** [comment]
@vladimir-kotikov two things:
1. The appveyor builds aren't working. It looks like fixing that requires logging into your appveyor account. Can you email me at my profile's email address, and we can chat about how best to update appveyor moving forward?  (E.g. maybe copy the appveyor config details into email, disable it from your appveyor account, then I could make my own appveyor account and set up a new config?)
2. Also, is there a way to grant me access to change Settings in the repo?

**Comment by vladimir-kotikov** [comment]
For the 1 - Yeah, let's figure that out, but honestly I think there's no much sense in keeping Appveyor as CI engine. Initially I thought I need to stick to Windows as it's the platform the completions code is running on, but I think it can be run on any other platform as long as there's a Lua exists for it. That said I think it might make sense to switch to Gitlab Actions which might be easier to manage. Doesn't look like a priority tho

For 2 - let me see, I'll try to figure out how to grant you all required permissions

**Comment by chrisant996** [comment]
@vladimir-kotikov how have you been merging from `dev` into `master`?

I used GitHub to make a pull request, and accepted the default "squash merge" action.

That seems to cause problems for subsequent merges and pull requests, so I want to double check how you've been doing things.

- What was your sequence of operations for merging master into dev?
- What was your sequence of operations for merging dev into master?

**Comment by vladimir-kotikov** [comment]
I haven't been using separate master and dev branches - IMHO in such a simple projects it would bring more maintenance overhead and complications than benefits. That said my flow usually was to create a feature branches and ff-merge them to main branch as soon as the feature complete, rebasing when master's HEAD is changed (e.g. when something else is merged) and interactively - right before merging, to clean commit history, squash something, etc. Releases were made simply by tagging commits from master.

The main reason for this flow was that I personally like when the commit history is readable and every commit signifies some isolated chunk of work or completed feature so you can clearly see what has been done. Basically the commit history was some sort of RELEASENOTES.md (which I actually been generating from history)

**Comment by DRSDavidSoft** [comment]
> The main reason for this flow was that I personally like when the commit history is readable and every commit signifies some isolated chunk of work or completed feature so you can clearly see what has been done. Basically the commit history was some sort of RELEASENOTES.md (which I actually been generating from history)

I also agree with this point, as squashing the commits essentially removes any commit history messages and instead turns all file changes into one big blob. 👍

**Comment by chrisant996** [comment]
It sounds like the `dev` branch is not where to stage changes, and branches should be used for only one pull request and then deleted.

I'll update the documentation accordingly, and use that model moving forward.

Thanks for clarifying which branching model this repo has been using.

**Comment by chrisant996** [comment]
> > The main reason for this flow was that I personally like when the commit history is readable and every commit signifies some isolated chunk of work or completed feature so you can clearly see what has been done. Basically the commit history was some sort of RELEASENOTES.md (which I actually been generating from history)
> 
> I also agree with this point, as squashing the commits essentially removes any commit history messages and instead turns all file changes into one big blob. 👍

It also can make regression investigations later on much more difficult, or less informative.

That branching model has benefits.  It also has drawbacks.  I'm willing to continue using that model here, since this is a tiny project.
